apps/web/src/components/Onboarding/SetupPassword/SetupPassword.test.tsx

```diff
@@ -0,0 +1,201 @@
+import {
+  type UmamiStore,
+  makeStore,
+  useRestoreFromMnemonic,
+  useRestoreFromSecretKey,
+} from "@umami/state";
+import { mnemonic1 } from "@umami/test-utils";
+
+import { CURVES, SetupPassword } from "./SetupPassword";
+import { act, fireEvent, renderInModal, screen, userEvent, waitFor } from "../../../testUtils";
+
+jest.mock("@umami/state", () => ({
+  ...jest.requireActual("@umami/state"),
+  useRestoreFromMnemonic: jest.fn(),
+  useRestoreFromSecretKey: jest.fn(),
+}));
+
+const password = "password";
+
+describe("<SetupPassword />", () => {
+  describe.each(["mnemonic", "secret_key"] as const)("%s mode", mode => {
+    describe("validations", () => {
+      it("requires a password", async () => {
+        await renderInModal(<SetupPassword mode={mode} />);
+
+        const passwordInput = screen.getByLabelText("Set Password");
+        fireEvent.click(passwordInput);
```

Comment:
Could we replace fireEvent with userEvent in places where it is possible like click, type?

Reply:
makes sense 👍